packages/nextjs/src/types.ts

```diff
@@ -13,7 +12,4 @@ export type NextClerkProviderProps = {
    * @default true
    */
   __unstable_invokeMiddlewareOnAuthStateChange?: boolean;
-} & Omit<IsomorphicClerkOptions, keyof PublishableKeyOrFrontendApi> &
-  Partial<PublishableKeyOrFrontendApi> &
-  Omit<IsomorphicClerkOptions, keyof MultiDomainAndOrProxy> &
-  MultiDomainAndOrProxy;
```

Comment:
🚫 I don't think we want to drop these

Reply:
Definition of `IsomorphicClerkOptions` [ref](https://github.com/clerk/javascript/blob/492b8a7b12f14658a384566012e5807f0a171710/packages/react/src/types.ts#L27) 
```
// TODO(@dimkl): Remove frontendApi when it's removed from ClerkOptions in @clerk/types
export type IsomorphicClerkOptions = Omit<ClerkOptions, 'isSatellite' | 'frontendApi'> & {
  Clerk?: ClerkProp;
  clerkJSUrl?: string;
  clerkJSVariant?: 'headless' | '';
  clerkJSVersion?: string;
  sdkMetadata?: SDKMetadata;
} & PublishableKeyOrFrontendApi &
  MultiDomainAndOrProxy;
```
I think we are just omitting `MultiDomainAndOrProxy` and adding `MultiDomainAndOrProxy` in this part of the code. It's just a refactoring that could be done outside of the context of this cleanup.

Reply:
I think this was necessary in order to make typescript happy with the complex types what we had due to our unions for PublishableKeyOrFrontendApi and MultiDomainAndOrProxy. Now that we would have only we should be ok.
We also have tests for those so the test should catch anything bad.